src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -54,7 +71,7 @@ public PreparedQuery getBigBusinessFromDatabase(){
   }
 
   public List<Listing> removeBigBusinessesFromResults(PreparedQuery queryOfDatabase){
-    Iterator<Listing> businesses =  allBusinesses.iterator();
+    businesses =  allBusinesses.iterator();
```

Comment:
I can not call allBusinesses.iterator().next() within a while loop, as the program will not go to the next element of the list. Instead as seen in all these examples: https://www.geeksforgeeks.org/how-to-use-iterator-in-java/. This is because, every time you call .iterator method, you are creating a new iterator, which results in only getting the first element of the list.

Reply:
What error are you getting when you try? According to the doc: https://docs.oracle.com/javase/8/docs/api/java/util/List.html this is supported. Even so, if you're only iterating through the list, a foreach loop will give you the same result with less boilerplate.

Reply:
If you look at this comment I just made, it has never worked for me. It does not seem you can't remove an element within a loop. https://github.com/googleinterns/SmallCity/pull/39#discussion_r461176814

Reply:
See my comment below on this.